extensions/shared/notification-on-plan-upgrade.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * External dependencies
+ */
+import { dispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { parse as parseUrl } from 'url';
+
+/**
+ * Shows a notification when a plan is marked as purchased
+ * after redirection from WPCOM.
+ */
+
+if ( undefined !== typeof window && window.location ) {
+	const { query } = parseUrl( window.location.href, true );
+
+	if ( query.plan ) {
```

Comment:
Just pointing out that the name of this query arg is going to become somewhat of a fixed API -- it's the redirect arg that we'll use in Jetpack to enable the nudge, and that Calypso will need to [whitelist](https://github.com/Automattic/wp-calypso/blob/a9d8ee1e6b3e32deb2521ca4a80e1279c7619088/client/my-sites/checkout/checkout/index.jsx#L422-L425). This means:

- it shouldn't change (much) across Jetpack releases, so there's not a number of different qargs out in the wild doing the same (or worse, slightly different things), and thus
- it should be a descriptive, ideally somewhat unambiguous name -- also for the sake of finding it easily in the codebase.

Since we're also going to use this from the upgrade nudge (#13218), I think it would also make sense to define it as a `const` in a shared file and import that from both places that use it.

I see that #13218 sets `plan=updated`. This is fairly semantic, but if we do that, I think we should reflect it here (the underlying principle vaguely being: don't over-define what you're not going to actually evaluate).

```suggestion
	if ( query.plan === 'updated' ) {
```

(I was originally assuming we'd just set it to `1` which IIRC is the query string way of saying that something is `true`.)

Alternatively, we could use something like `upgrade=1`, `upgraded=1`, `plan_upgrade=1`, `show_upgrade_notice=1` (the latter strongly implying the UI effect, thus maybe too specific in case we want to change this at some point).

Note that none of this hairsplitting is supposed to block this PR :sweat_smile: I'd like to see the shared constant at least, but fine to do in a follow-up.

Reply:
Thanks! Agreed on all points, and thanks for the extra info. Kinda hard to choose there though! I think i like some form of "upgraded" as it signals an action that "has happened". I'd perhaps go with "plan_upgraded" instead, but not sure. Any preference? (happy to handle all these here)

Reply:
No preference on my part, I'm fine with `plan_upgraded` :slightly_smiling_face:

Reply:
p.s. I think the intention with the original form was to also pass the slug back/forth i.e. `&plan=jetpack_premium` that we can match against the list of plans in upgrade-nudge. But yeah, I'd assume a separate auth API call for site data would be best.

Reply:
Yeah, we should be able to gather that information easily from within Jetpack. I'd rather not rely on query args too much to convey information since they're not that reliable after all.